File: haversine.js
var EARTH_RADIUS = 6371;
var toRad = function (degrees) { return (degrees * Math.PI) / 180; };
// https://en.wikipedia.org/wiki/Haversine_formula
var haversine = function (from, to) {
    var dLat = toRad(to[0] - from[0]);
    var dLon = toRad(to[1] - from[1]);
    var theta = toRad(from[0]);
    var phi = toRad(to[0]);
    var a = Math.pow(Math.sin(dLat / 2), 2) +
        Math.pow(Math.sin(dLon / 2), 2) * Math.cos(theta) * Math.cos(phi);
    console.log(a);
    var c = 2 * Math.asin(Math.sqrt(a));
    return EARTH_RADIUS * c;
};
console.log(haversine([53, -6], [43, 2]));

File: haversine.c
#include <stdio.h>
#include <math.h>

static const int EARTH_RADIUS = 6371;

double toRad(double d)
{
  return (d * M_PI) / 180;
}

double haversine(double from[2], double to[2])
{
  double dLat = toRad(to[0] - from[1]);
  double dLon = toRad(to[1] - from[1]);

  double theta = toRad(from[0]);
  double phi = toRad(to[0]);

  double a = pow(sin((dLat / 2)), 2) + pow(sin(dLon / 2), 2) * cos(theta) * cos(phi);

  double c = 2 * asin(sqrt(a));

  return c * EARTH_RADIUS;
}

int main()
{
  double from[2] = {53, -6};
  double to[2] = {43, 4};

  double d = haversine(from, to);
  printf("%f", d);

  return 0;
}
